maestro-ios-xctest-runner/maestro-driver-iosUITests/Routes/Handlers/SessionInfoHandler.swift

```diff
@@ -0,0 +1,33 @@
+import Foundation
+import FlyingFox
+import os
+import XCTest
+import Network
+
+@MainActor
+struct SessionInfoHandler: HTTPHandler {
+
+    let logger = Logger(
+        subsystem: Bundle.main.bundleIdentifier!,
+        category: String(describing: Self.self)
+    )
+    
+    private let sessionId = UUID()
```

Comment:
Why is the XCTest runner involved to create a UUID? Seems like this can be done much faster and simpler in the worker..?

Reply:
When the user uses the maestro SDK to use the Mock Server feature (https://maestro.mobile.dev/advanced/experimental/maestro-mock-server/getting-started#setup), it will proxy the app requests through our backend. The backend stores the events and then the maestro client needs to check if the data used in the `assertOutgoingRequest` matches the info stored.

The sessionId is related to a specific maestro session, meaning every time it runs `maestro test <flow>`. Because of that, the driver should create a new UUID every time and then, the client will get the sessionId and fetch the backend to get the events. That's why we can't use worker for that.

Reply:
Sounds like XCTest runner is really not the right place for that. It can be restarted throughout the flow and should not be used for storing UUID's. Can't maestro store the session UUID?

Reply:
How about the SDK doing a call-out to our backend with the sessionId to match on as a HTTP call?

Reply:
@berikv the request to our backend with the sessionId to match on HTTP call happens from maestro and not from the SDK, as maestro needs the response in order to validate the assertions. Is there any way we can generate a session id from within the iOS SDK and keep that session id throughout the flow execution?

Reply:
What is a session exactly? How long does it live? Where does it depend on?

* If it is a flow run, then maestro should be the source of truth, eg Orchestra or SessionManager could keep this state.
* If a session is longer than a single flow run, then the source of truth needs to be outside of maestro (and not in the XCTest runner, which is part of maestro).
 * If a session is bound by the SDK, then let the SDK be the source of truth. If maestro needs to know the session ID, it can fetch it using HTTP calls from the backend or directly to the app, whatever works.
* If it is just a part of a flow run, the source of truth can still be maestro. A session object that's closed and reopened when needed. It could live alongside Orchestra.